pkg/reconciler/taskrun/taskrun.go

```diff
@@ -186,6 +187,11 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, tr *v1beta1.TaskRun) pkg
 	// updates regardless of whether the reconciliation errored out.
 	if err = c.reconcile(ctx, tr, rtr); err != nil {
 		logger.Errorf("Reconcile: %v", err.Error())
+		if errors.Is(err, sidecarlogresults.ErrorReasonMaxResultSizeExceeded) {
+			message := fmt.Sprintf("TaskRun %q failed to finish because atleast one of its results exceeded the max allowed limit.", tr.Name)
```

Comment:
Prefer brevity for readability: "TaskRun %q failed: results exceeded size limit %d bytes"